### DISCUSSION TITLE
Should we consider adding a `ISecretProvider` for the command line arguments?

### DISCUSSION THREAD

**Discussion by stijnmoreels** [arcus-azure/arcus.security#238] [Ideas]
It may be useful to add a `ISecretProvider` for command line arguments? Especially in an application which is controlled from the command line.

**Comment by tomkerkhove** [comment]
What applications do you have in mind?

**Reply by stijnmoreels** [reply]
Console applications, like the application we have to set the labels on our GH repo's. It uses secrets and bc it's a rather small application it's OK to use it directly, but in a larger one, it might be useful to have a secret provider that takes in the command line arguments. Only not sure if we should support this by default.

**Reply by tomkerkhove** [reply]
I don't think we'll have many users that will use it but seems fairly easy to add so happy to include it
